### PR TITLE
docs:  Add entry for backslash_quotes liberal parsing

### PR DESCRIPTION
**Pull request by markaschneider**

### CODE REVIEW

File: doc/csv/options/parsing/liberal_parsing.rdoc [hunk 1]
```diff
@@ -17,3 +17,22 @@ Without +liberal_parsing+:
...
+
+With +liberal_parsing+, but without the +backslash_quote+ sub-option:
+  # Incorrect interpretation of backslash; incorrectly interprets the quoted comma as a field separator.
+  ary = CSV.parse_line(str, liberal_parsing: true)
+  ary # => ["Show", "\"Harry \\\"Handcuff\\\" Houdini", " the one and only\"", "Tampa Theater"]
+  puts ary[1] # => "Harry \"Handcuff\" Houdini"
+
+With +liberal_parsing+ and its +backslash_quote+ sub-option:
+  ary = CSV.parse_line(str, liberal_parsing: { backslash_quotes: true })
+  ary # => ["Show", "Harold \"Handcuff\" Houdini", "Tampa Theater"]
```
[the remark this answers is not in the document]
Comment (markaschneider) [reply doc/csv/options/parsing/liberal_parsing.rdoc:37]: @kou Thanks for catching that.  I thought I had copied and pasted, but apparently not.  `backslash_quotes: true` should be `backslash_quote: true`.  I'll resubmit.

[the remark this answers is not in the document]
Comment (markaschneider) [reply doc/csv/options/parsing/liberal_parsing.rdoc:37]: Here's a console log:

```
 ~/code/csv/doc/csv/options/parsing git:(doc/add-liberal-parsing) docker run -it 1e6b2a26f38d bash
root@d277469b2d78:/# irb
irb(main):001:0> require 'csv'
=> true

irb(main):002:0> RUBY_VERSION
=> "3.2.0"

irb(main):003:0> str = 'Show,"Harry \"Handcuff\" Houdini, the one and only","Tampa Theater"'
=> "Show,\"Harry \\\"Handcuff\\\" Houdini, the one and only\",\"Tampa Theater\""

irb(main):004:0> ary = CSV.parse_line(str, liberal_parsing: true)
=> ["Show", "\"Harry \\\"Handcuff\\\" Houdini", " the one and only\"", "Tampa Theater"]

irb(main):005:0> puts ary[1]
"Harry \"Handcuff\" Houdini
=> nil

irb(main):006:0>

irb(main):007:0> ary = CSV.parse_line(str, liberal_parsing: { backslash_quote: true })
=> ["Show", "Harry \"Handcuff\" Houdini, the one and only", "Tampa Theater"]

irb(main):008:0> puts ary[1]
Harry "Handcuff" Houdini, the one and only
=> nil
```

[the remark this answers is not in the document]
Comment (kou) [reply doc/csv/options/parsing/liberal_parsing.rdoc:37]: Could you also update the results?

[the remark this answers is not in the document]
Comment (markaschneider) [reply doc/csv/options/parsing/liberal_parsing.rdoc]: Sorry for the delay.  I resubmitted just now.

**Comment by kou** [comment]
Thanks!

**Pull request merged**